packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -116,6 +123,36 @@ if (__DEV__) {
   };
 }
 
+// Context (new API)
+let providerStack: Array<ReactProvider<mixed>> = []; // Stack of provider objects
```

Comment:
Is this reentrant?

Imagine `renderToString(<App />)` where `App` calls `renderToString()` on another tree.

Reply:
Ooh, yeah, I'll put these on the instance and add a test